### PR TITLE
Config

### LINKED ISSUE #4: Config

### PR DESCRIPTION
**Pull request by GuidoTournois**
Fixes #4

### CODE REVIEW

File: {{cookiecutter.project_slug}}/config.ini [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+[DEVELOPMENT]
+data_path = "~/data"
+
+
+[TEST]
+data_path = "/TEST/data/"
+
+
+[PRODUCTION]
+data_path = "/PRD/data/"
```
Comment (gapper100) [inline {{cookiecutter.project_slug}}/config.ini:16]: Does it make sense to have DEVELOPMENT in the 'root', and TEST and PRODUCTION in subfolders?

File: {{cookiecutter.project_slug}}/config.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import os
+import ast
+import sys
+import configparser
+from typing import Dict
+from dataclasses import dataclass
+from typing import Dict
```
Comment (gapper100) [inline {{cookiecutter.project_slug}}/config.py:7]: Duplicate import

File: {{cookiecutter.project_slug}}/config.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import os
+import ast
+import sys
```
Comment (gapper100) [inline {{cookiecutter.project_slug}}/config.py:3]: I believe this isn't used

File: {{cookiecutter.project_slug}}/config.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import configparser
+from typing import Dict
+from dataclasses import dataclass
+from typing import Dict
+
+
+class Singleton(type):
+    __instances: Dict = {}
```
Comment (gapper100) [inline {{cookiecutter.project_slug}}/config.py:11]: No/different type of DocString

File: {{cookiecutter.project_slug}}/config.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+@dataclass
+class Config(metaclass=Singleton):
+    """
+    Class which contains our configuration. Once initialized, it cannot be changed, as it is a singleton.
+    """
+    param1 : str
+    param2 : int
+    data_path : str
```
Comment (gapper100) [inline {{cookiecutter.project_slug}}/config.py:35]: Perhaps remove this? Pattern is clear from the other two.

File: {{cookiecutter.project_slug}}/config.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    def __init__(
+        self,
+        env: str = "DEVELOPNMENT",
```
Comment (gapper100) [inline {{cookiecutter.project_slug}}/config.py:39]: Typo: DEVELOPMENT

File: {{cookiecutter.project_slug}}/config.ini (hunk 1 above)
Comment (GuidoTournois) [reply {{cookiecutter.project_slug}}/config.ini:16]: No it does not, what would you propose for example variables in the config?

File: {{cookiecutter.project_slug}}/config.py (hunk 2 above)
Comment (GuidoTournois) [reply {{cookiecutter.project_slug}}/config.py:7]: Such an observant eye

File: {{cookiecutter.project_slug}}/config.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        """
+        if project_root:
+            self.project_root = project_root
+        else:
+            self.project_root = os.path.dirname(os.path.abspath(__file__))
+
+        if not os.path.exists(self.project_root):
+            raise FileNotFoundError("The path given as project root does not exist")
+        if not os.path.isdir(self.project_root):
+            raise NotADirectoryError("The path given is not a directory, but a file path")
```
Comment (GuidoTournois) [inline {{cookiecutter.project_slug}}/config.py:57]: Add exceptions when wrong input is given for project root

**Pull request merged**